tests/standalone_plugins/test_file_extensions.py

```diff
@@ -0,0 +1,77 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2024 Greenbone AG
+import sys
+import tempfile
+import unittest
+from argparse import Namespace
+from io import StringIO
+from pathlib import Path
+from unittest.mock import patch
+
+from troubadix.standalone_plugins.file_extensions import (
+    check_extensions,
+    main,
+    parse_args,
+)
+
+
+class TestFileExtensions(unittest.TestCase):
+    def test_ok(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            tempfile.mkstemp(dir=tmpdir, suffix=".nasl")
+            tempfile.mkstemp(dir=tmpdir, suffix=".inc")
+            child_dir = tempfile.mkdtemp(dir=tmpdir)
+            tempfile.mkstemp(dir=child_dir, suffix=".nasl")
+            parsed_args = Namespace(dirs=[Path(tmpdir)])
+            self.assertFalse(check_extensions(parsed_args))
+
+    def test_fail(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            # foo.nasl.nasl
+            fp1 = Path(tempfile.mkstemp(dir=tmpdir, suffix=".nasl.nasl")[1])
+            # foo.inc.inc
+            fp2 = Path(tempfile.mkstemp(dir=tmpdir, suffix=".inc.inc")[1])
+            # foo.nasl.inc
+            fp3 = Path(tempfile.mkstemp(dir=tmpdir, suffix=".nasl.inc")[1])
+            # foo.inc.nasl
+            fp4 = Path(tempfile.mkstemp(dir=tmpdir, suffix=".inc.nasl")[1])
+            child_dir = tempfile.mkdtemp(dir=tmpdir)
+            # foo.bar
+            fp5 = Path(tempfile.mkstemp(dir=child_dir, suffix=".bar")[1])
+            # .foo
+            fp6 = Path(tempfile.mkstemp(dir=child_dir, prefix=".")[1])
+            # foo
+            fp7 = Path(tempfile.mkstemp(dir=child_dir)[1])
```

Comment:
Not fully sure how these tests are working but would it make sense to check each of these in the `tmpdir` and in the `child_dir` if not already the case?

Reply:
Toplevel and subdir file paths are all covered by the same call to Path.rglob(*). If one works, so will the others. Testing for both is therefore unnecessary, but I have tested a few just to make sure that rglob behaves as I would expect.

Reply:
Thanks for clarification and would be fine for me then. But i would leave this up to one of the other @greenbone/troubadix-maintainers to solve this thread if they are fine as well :+1: